java/parser.js

```diff
@@ -1,171 +1,206 @@
+    if (cs.matchLevel) options += `.matchLevel(MatchLevel.${cs.matchLevel.toUpperCase()})`;
+    if (cs.name) options += `.withName("${cs.name}")`;
+    if (cs.layoutBreakpoints) options += `.layoutBreakpoints(${cs.layoutBreakpoints})`;
     if (cs.isFully === true) {
-        options += '.fully()'
+        options += '.fully()';
     } else if (cs.isFully === false) {
-        options += '.fully(false)'
+        options += '.fully(false)';
+    }
+    if (cs.visualGridOptions) {
+        const VGOptionsKeys = Object.keys(cs.visualGridOptions);
+        const vgOptions = VGOptionsKeys.map(key => `new VisualGridOption("${key}", ${cs.visualGridOptions[key]})`).join(', ');
+        options += `.visualGridOptions(${vgOptions})`;
+    }
+    if (cs.variationGroupId) options += `.variationGroupId("${cs.variationGroupId}")`;
+    if (cs.hooks) {
+        checkOptions(cs.hooks, CHECK_SETTINGS_HOOKS);
+        if (cs.hooks.beforeCaptureScreenshot) {
+            options += `.beforeRenderScreenshotHook("${cs.hooks.beforeCaptureScreenshot}")`;
+        }
     }
-    if (cs.variationGroupId) options += `.variationGroupId("${cs.variationGroupId}")`
-    return java + element + options
+	if (cs.pageId) {
+		options += `.pageId("${cs.pageId}")`;
+	}
+    return java + element + options;
 
     // check settings
 
     function frames(arr) {
-        return arr.reduce((acc, val) => acc + `${frame(val)}`, '')
+        return arr.reduce((acc, val) => acc + `${frame(val)}`, '');
     }
 
     function frame(frame) {
-        return (!frame.isRef && frame.frame) ? `.frame(${frameSelector(frame.frame)}).scrollRootElement(${printSelector(frame.scrollRootElement)})` : `.frame(${frameSelector(frame)})`
+        return (!frame.isRef && frame.frame) ? `.frame(${frameSelector(frame.frame)}).scrollRootElement(${printSelector(frame.scrollRootElement)})` : `.frame(${frameSelector(frame)})`;
     }
 
     function frameSelector(selector) {
         if (typeof selector === 'string' && !checkCss(selector)) {
-            return JSON.stringify(selector)
+            return JSON.stringify(selector);
         } else {
             return printSelector(selector);
         }
 
         function checkCss(string) {
-            return (string.includes('[') && string.includes(']')) || string.includes('#')
+            return (string.includes('[') && string.includes(']')) || string.includes('#');
         }
     }
 
     function region(region) {
-        return `.region(${regionParameter(region)})`
+        return `.region(${regionParameter(region)})`;
     }
 
     function ignoreRegions(arr) {
-        return arr.reduce((acc, val) => `${acc}.ignore(${regionParameter(val)})`, '')
+        return arr.reduce((acc, val) => `${acc}.ignore(${regionParameter(val)})`, '');
     }
 
     function layoutRegions(arr) {
-        return arr.reduce((acc, val) => `${acc}.layout(${regionParameter(val)})`, '')
+        return arr.reduce((acc, val) => `${acc}.layout(${regionParameter(val)})`, '');
```

Comment:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/applitools/sdk.coverage.tests/pullRequest?prid=8528488)
```suggestion
        return arr.reduce((acc, val) => `${acc}.layout(${regionParameter(val)})`, "");
```